src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1123,6 +1123,10 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                     scriptConfig.Functions.Add((string)function);
                 }
             }
+            else
+            {
+                scriptConfig.Functions = null;
```

Comment:
The biggest issue is that we end up modifying the state (the config object) on initialization. I'm working on some changes to prevent that.